### DISCUSSION TITLE
Show only specific dates and ability to show recurring dates

### DISCUSSION THREAD

**Discussion by sujit-baniya** [gpbl/react-day-picker#1965] [Ideas]
> Feature Request: 1

It would be less code on developer's side to just show the specific dates and hiding others. Currently I could use `hidden` to hide all non-essential dates but that would get a lot date ranges to hide. 
So a property like `show: [2023-01-01]` would be good to show only that date hiding other dates

> Feature Request: 2

I could select recurring day per week or dates per month and every other dates would be hidden/disabled.
For e.g. `Weekly Sun, Tue Wed` would allow me to only show Sun Tue and Wed for every week

**Comment by gpbl** [comment]
Hi @sujit-baniya thanks for your suggestions1

> It would be less code on developer's side to just show the specific dates and hiding others. 

Curious to see your code. `hidden` can get a function too, so that you can actually _negate_ the `hidden` prop:

```tsx
// Suppose `daysToShow` is the array of the days you want to display:

<DayPicker hidden={ date => { return !daysToShow.includes(date); } } />
```

> I could select recurring day per week or dates per month and every other dates would be hidden/disabled.
For e.g. Weekly Sun, Tue Wed would allow me to only show Sun Tue and Wed for every week

Same as above, you can use a function:

```tsx
// Will hide all Sun for every week;

<DayPicker hidden={ date => { return date.getDay() === 0; } } />
```

**Comment by ashrafnazar** [comment]
Hi @gpbl - can you hide weekends? So on the calendar, the user will only be able to see Mon-Fri

**Reply by ashrafnazar** [reply]
I managed to do partially what I wanted using the prop, whilst rendering the weekend day columns. See screenshot attached (I don't want anything in the red rectangles to render)

<img width="328" alt="Screenshot 2024-01-03 at 17 25 00" src="https://github.com/gpbl/react-day-picker/assets/148764477/6a31ea91-9730-433e-a100-f8f85070371d">

**Reply by gpbl** [reply]
@ashrafnazar you can also remove them via CSS - would that work for you?

https://codesandbox.io/p/sandbox/react-day-picker-forked-sszl6j?file=%2Fsrc%2Fstyles.css


```css
.rdp-head_cell:first-child,
.rdp-head_cell:last-child,
.rdp-cell:first-child,
.rdp-cell:last-child {
  display: none;
}
```

**Reply by ashrafnazar** [reply]
@gpbl - Thanks for the reply!

Would it be worthwhile, having a prop to handle this? Would be nice if no CSS is required and the HTML markup doesn't feature the table header at all. Am keen to hear what you think

**Reply by ashrafnazar** [reply]
If you would consider this, I'm happy to work on a PR - just need a little direction on what I should change, as I'm not really familiar with the repo
